### PR TITLE
control_valid_metric

### PR DESCRIPTION
**Pull request by MariamTurky**

### CODE REVIEW

File: liver_imaging_analysis/config/configs.json [hunk 1]
```diff
@@ -58,5 +55,33 @@
...
+        "reference_liver_2d_array": "tests/testdata/testcases/predicted_3d.npy",
+        "reference_liver_sw_array": "tests/testdata/testcases/predicted_sliding.npy",
+        "reference_lobe_array": "tests/testdata/testcases/predicted_lobe.npy",
+        "reference_gif": "tests/testdata/testcases/gif_frames",
+        "liver_slices": "tests/testdata/testcases/liverSlices",
+        "liver_png_slices": "tests/testdata/testcases/liverSlices_png",
+        "reference_batches": "tests/testdata/testcases/batches"
+    },
+    "tune": {
+        "check_point": "/content/drive/MyDrive/BMATM_LiverSeg_2023/HMTYB_Liver_Seg_2024/experimental/tune_check_points/1"
```
Comment (Muhammad-Al-Barbary) [inline liver_imaging_analysis/config/configs.json:80]: absolute paths shouldn't be here

File: liver_imaging_analysis/engine/engine.py [hunk 2]
```diff
@@ -338,6 +338,7 @@ def fit(
         batch_callback_epochs = None,
         save_weight = False,
         save_path = config.save["potential_checkpoint"],
+        metric_epoch =1
```
Comment (Muhammad-Al-Barbary) [inline liver_imaging_analysis/engine/engine.py:341]: why are you adding another parameter for metric? can't you use evaluate_epochs for this purpose?

File: liver_imaging_analysis/engine/engine.py [hunk 3]
```diff
@@ -394,22 +395,38 @@ def fit(
...
+                        metric_epoch,
+                    )
+            else:
+                self.per_epoch_callback(
                     epoch,
                     training_loss,
                     valid_loss,
                     training_metric,
-                    valid_metric,
+                    None,
```
Comment (Muhammad-Al-Barbary) [inline liver_imaging_analysis/engine/engine.py:424]: why would you need to do this?

File: liver_imaging_analysis/engine/engine.py [hunk 4]
```diff
@@ -443,9 +460,11 @@ def test(self, dataloader = None, callback = False):
                     batch[Keys.PRED],
                     batch[Keys.LABEL]
                     ).item()
-                #Apply post processing transforms on prediction
-                batch = self.post_process(batch)
-                self.metrics(batch[Keys.PRED].int(), batch[Keys.LABEL].int())
+                
+                if post_process:
```
Comment (Muhammad-Al-Barbary) [inline liver_imaging_analysis/engine/engine.py:464]: maybe we should separate mandatory post processing (activation/discretization) from enhancement post processing. It doesn't make sense to skip metric calculation if you don't want to post process the model outputs

File: liver_imaging_analysis/models/liver_segmentation.py [hunk 5]
```diff
@@ -717,39 +720,32 @@ def train_liver(
...
         batch_callback_epochs = batch_callback_epochs,
         save_weight = save_weight,
-        save_path = save_path
+        save_path = save_path,
+        metric_epoch = metric_epoch
     )
     # Evaluate on latest saved check point
     model.load_checkpoint(save_path)
-    final_loss, final_metric = model.test(
+    final_loss,_ = model.test(
```
Comment (Muhammad-Al-Barbary) [inline liver_imaging_analysis/models/liver_segmentation.py:741]: something wrong in this line, are you returning  the metric or not?

**Review by Muhammad-Al-Barbary — COMMENTED** [review]
I don't get the logic of this PR. What is the purpose of these modifications?

File: liver_imaging_analysis/config/configs.json (hunk 1 above)
Comment (MariamTurky) [reply liver_imaging_analysis/config/configs.json]: done

File: liver_imaging_analysis/engine/engine.py (hunk 3 above)
Comment (MariamTurky) [reply liver_imaging_analysis/engine/engine.py:423]: because this error : local variable 'valid_metric' referenced before assignment

**Review by helrewaidy — CHANGES_REQUESTED** [review]
I also don't get the point of this PR!
You should compute the metric with or without post-processing, right? if so, please revert this PR

**Pull request closed without merging**